### PR TITLE
[RESTEASY-2061] exclude jaxb-api archive from resteasy-multipart-prov…

### PR DESCRIPTION
**Pull request by rsearls**
…ider and resteasy-jaxb-provider

### CODE REVIEW

File: resteasy-bom/pom.xml [hunk 1]
```diff
@@ -40,6 +40,12 @@
                 <groupId>org.jboss.resteasy</groupId>
                 <artifactId>resteasy-jaxb-provider</artifactId>
                 <version>${project.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>javax.xml.bind</groupId>
+                        <artifactId>jaxb-api</artifactId>
+                    </exclusion>
+                </exclusions>
```
Comment (asoldano) [inline resteasy-bom/pom.xml]: Why not excluding this for the jaxb-runtime dependency in resteasy-dependencies/pom.xml  ?

Comment (rsearls) [reply resteasy-bom/pom.xml]: This is no jaxb-runtime in the BOM.  I checked that first.

Comment (asoldano) [reply resteasy-bom/pom.xml]: https://github.com/resteasy/Resteasy/blob/master/resteasy-dependencies-bom/pom.xml#L225 ?

Comment (jbliznak) [reply resteasy-bom/pom.xml]: @asoldano what is the purpose to have two distinct BOMs anyway? 
https://github.com/resteasy/Resteasy/blob/master/resteasy-bom/pom.xml
https://github.com/resteasy/Resteasy/blob/master/resteasy-dependencies-bom/pom.xml

If used for differrent purposes then they both should handle excludes in the same way, right?

**Pull request merged**